odoo/tools/translate.py

```diff
@@ -888,6 +889,45 @@ def handle_text(text, lineno):
     _extract_translatable_qweb_terms(tree.getroot(), handle_text)
     return result
 
+
+def extract_spreadsheet_terms(fileobj, keywords, comment_tags, options):
+    """Babel message extractor for spreadsheet data files.
+
+    :param fileobj: the file-like object the messages should be extracted from
+    :param keywords: a list of keywords (i.e. function names) that should
+                     be recognized as translation functions
+    :param comment_tags: a list of translator tags to search for and
+                         include in the results
+    :param options: a dictionary of additional options (optional)
+    :return: an iterator over ``(lineno, funcname, message, comments)``
+             tuples
+    :rtype: Iterable
+    """
+    terms = []
+    data = json.load(fileobj)
+    for sheet in data.get('sheets', []):
+        for cell in sheet['cells'].values():
+            content = cell.get('content', '')
+            terms += re.findall(r'_t\("(.+?)"\)', content, re.IGNORECASE)
+            markdown_link = re.fullmatch(r'\[(.+)\]\(.+\)', content)
```

Comment:
So a non-formula cell can only be composed of a single link, with no other data, and especially not 2+ links?

Reply:
non-formula cells can be any string.
A (single) markdown link is interpreted as a "link" cell. And you can't have 2+ links

Reply:
> And you can't have 2+ links

![nojoy](https://user-images.githubusercontent.com/7571158/192209867-ead39f00-0812-4b04-b2ca-40a3f19788eb.jpg)